File: payroll-api/src/main/java/com/company/payroll/service/impl/StaffBankingInfoServiceImpl.java
package com.company.payroll.service.impl;

import java.util.Optional;

import com.company.payroll.model.StaffBankingInfo;
import com.company.payroll.service.StaffBankingInfoService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.company.payroll.mapper.StaffBankingInfoMapper;
import com.github.pagehelper.PageHelper;
import com.github.pagehelper.PageInfo;

@Service
public class StaffBankingInfoServiceImpl implements StaffBankingInfoService {

	@Autowired
	private StaffBankingInfoMapper staffBankingInfoMapper;
	
	@Override
	public Optional<StaffBankingInfo> findById(Integer bId) {
		return Optional.ofNullable(staffBankingInfoMapper.selectByPrimaryKey(bId));
	}

	@Override
	public PageInfo<StaffBankingInfo> list(int page, int offset) {
		PageHelper.startPage(page, offset);
		return new PageInfo<StaffBankingInfo>(staffBankingInfoMapper.selectList());
	}

	@Override
	public StaffBankingInfo update(StaffBankingInfo bankingInfo) {
		staffBankingInfoMapper.updateByPrimaryKey(bankingInfo);
		return bankingInfo;
	}
}

File: kotlin-payroll/src/main/kotlin/com/company/payroll/service/impl/StaffBankingInfoServiceImpl.kt
package com.company.payroll.service.impl

import com.company.payroll.mapper.StaffBankingInfoMapper
import com.company.payroll.model.StaffBankingInfo
import com.company.payroll.service.StaffBankingInfoService
import com.github.pagehelper.PageHelper
import com.github.pagehelper.PageInfo
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class StaffBankingInfoServiceImpl(@Autowired private val staffBankingInfoMapper: StaffBankingInfoMapper): StaffBankingInfoService {
  override fun findById(bid: Int): StaffBankingInfo {
    return staffBankingInfoMapper.selectByPrimaryKey(bid)
  }

  override fun list(page: Int, offset: Int): PageInfo<StaffBankingInfo> {
    PageHelper.startPage<StaffBankingInfo>(page, offset)
    return PageInfo<StaffBankingInfo>(staffBankingInfoMapper.selectList())
  }

  override fun update(staffBankingInfo: StaffBankingInfo): StaffBankingInfo {
    staffBankingInfoMapper.updateByPrimaryKeySelective(staffBankingInfo)
    return staffBankingInfo
  }
}